src/commands.rs

```diff
@@ -558,38 +558,51 @@ struct SplitArgs {
 /// descendants, and `-r` to rebase a single commit. If none of them is
 /// specified, it defaults to `-b @`.
 ///
-/// With `-b`, it rebases the whole branch containing the specified revision.
-/// Unlike `-s` and `-r`, the `-b` mode takes the destination into account
-/// when calculating the set of revisions to rebase. That set includes the
-/// specified revision and all ancestors that are not also ancestors
-/// of the destination. It also includes all descendants of those commits. For
-/// example, `jj rebase -b B -d D` or `jj rebase -b C -d D`  would transform
-/// your history like this:
+/// With `-s`, the command rebases the specified revision and its descendants
+/// onto the destination. For example, `jj rebase -s C -d F` would transform
+/// your history like this (letters followed by an apostrophe are post-rebase
+/// versions):
 ///
-/// D          B'
-/// |          |
-/// | C        D
-/// | |   =>   |
-/// | B        | C'
-/// |/         |/
-/// A          A
+/// F           D'
+/// |           |
+/// | D         C'
+/// | |         |
+/// | C         F
+/// | |    =>   |
+/// | | E       | E
+/// | |/        | |
+/// | B         | B
+/// |/          |/
+/// A           A
 ///
-/// With `-s`, it rebases the specified revision and its descendants onto the
-/// destination. For example, `jj rebase -s C -d D` would transform your history
-/// like this:
+/// With `-b`, the command rebases the whole "branch" containing the specified
+/// revision. A "branch" is the set of commits that includes:
 ///
-/// D          C'
-/// |          |
-/// | C        D
-/// | |   =>   |
-/// | B        | B
-/// |/         |/
-/// A          A
+/// * the specified revision and ancestors that are not also ancestors of the
+///   destination
+/// * all descendants of those commits
+///
+/// In other words, `jj rebase -b X -d Y` rebases commits in the revset
+/// `(Y..X):` (which is equivalent to `jj rebase -s 'roots(Y..X)' -d Y` for a
+/// single root). For example, `jj rebase -b C -d F` or `jj rebase -b E -d F`
```

Comment:
two commands do the same thing? this is confusing

Reply:
It's also quite convenient :)

Reply:
maybe, if you understand it... if the graph is supposed to match both commands, perhaps say so? I don't think it's a good help text with confusing things in it...

Reply:
Ah, I see what you mean. Sure, that makes sense. See if this is better.

Reply:
I guess that's a little better. 
I wonder if you could change the examples to use a part of the alphabet that isn't also in the command options. Maybe J, K, L, M, N, O so there aren't two of some letters.

Reply:
Makes sense. Done.